packages/block-library/src/button/edit.js

```diff
@@ -133,10 +134,67 @@ function ButtonEdit( props ) {
 
 	const [ isEditingURL, setIsEditingURL ] = useState( false );
 	const isURLSet = !! url;
-	const opensInNewTab = linkTarget === NEW_TAB_TARGET;
-	const nofollow = !! rel?.includes( NOFOLLOW_REL );
+
 	const isLinkTag = 'a' === TagName;
 
+	// Defines how block attributes map to link value and vice versa.
+	const linkValueAttrsToDataMapping = {
+		url: {
+			dataKey: 'url',
+			toData: ( value ) => prependHTTP( value ),
+		},
+		opensInNewTab: {
+			dataKey: 'linkTarget',
+			toLink: ( value ) => value === NEW_TAB_TARGET,
+			toData: ( value ) => ( value ? NEW_TAB_TARGET : undefined ),
+		},
+		nofollow: {
+			dataKey: 'rel',
+			toLink: ( value ) => value?.includes( NOFOLLOW_REL ),
+			toData: ( value, { opensInNewTab: opensInNewTabValue } ) => {
+				// "rel" attribute can be effected by changes to
+				// "nofollow" and "opensInNewTab" attributes.
+				// In addition it is editable in plaintext via the UI
+				// so consider that it may already contain a value.
+				let updatedRel = '';
```

Comment:
@getdave I think we need to pass the current attributes value to this method in order taking the current value into account, I this example, it's the link `rel` attribute from Button edit. Maybe like
```diff
const { toLink, toData } = getLinkValueTransforms(
+  currentAttributeValues,
    linkValueAttrsToDataMapping
);
```
How do you think?

Reply:
I don't _think_ we will need to do this. Why? Because the `toData` function receives the current data as the 3rd argument of the callback.

The signature is, roughly:

```js
function toDataCallback(currentValue, fullValueObject, fullDataObject)
```

Reply:
Thank you! I missed this